vulnerability/events_creator.go

```diff
@@ -0,0 +1,143 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package vulnerability
+
+import (
+	"strings"
+	"time"
+
+	db_types "github.com/aquasecurity/trivy-db/pkg/types"
+	trivy_types "github.com/aquasecurity/trivy/pkg/types"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	libevents "github.com/elastic/beats/v7/libbeat/beat/events"
+	"github.com/elastic/cloudbeat/resources/providers/awslib/ec2"
+	"github.com/elastic/cloudbeat/transformer"
+	"github.com/elastic/cloudbeat/version"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Vulnerability struct {
+	Cvss           db_types.VendorCVSS  `json:"cvss,omitempty"`
+	DataSource     *db_types.DataSource `json:"data_source,omitempty"`
+	Scanner        Scanner              `json:"scanner,omitempty"`
+	Score          Score                `json:"score,omitempty"`
+	Package        Package              `json:"package,omitempty"`
+	Cwe            []string             `json:"cwe,omitempty"`
+	ID             string               `json:"id,omitempty"`
+	Title          string               `json:"title,omitempty"`
+	Enumeration    string               `json:"enumeration,omitempty"`
+	Reference      string               `json:"reference,omitempty"`
+	Description    string               `json:"description,omitempty"`
+	Severity       string               `json:"severity,omitempty"`
+	Classification string               `json:"classification,omitempty"`
+	PublishedDate  *time.Time           `json:"published_date,omitempty"`
```

Comment:
`Vulnerability` is missing `report_id` as described in the index.
It should have the same value as in `event.sequence`.